src/backend/cdb/motion/cdbmotion.c

```diff
@@ -1274,3 +1277,77 @@ UpdateSentRecordCache(int32 *sent_record_typmod)
 		*sent_record_typmod = NextRecordTypmod;
 	}
 }
+
+void
+SetCurrentMotionIPCLayer(int new_type)
+{
+	int type;
+
+	/* interconnect.so is not loaded. */
+	if (CurrentIPCLayerImplNum == 0)
+	{
+		Assert(CurrentMotionIPCLayer == NULL);
+		Gp_interconnect_type = new_type;
+		ereport(WARNING,
+				(errcode(ERRCODE_WARNING_GP_INTERCONNECTION),
+				 errmsg("No any IPCLayer implement loaded.")));
+
+		return;
+	}
+
```

Comment:
How to distinguish no ic is available from ic will load later?

Reply:
fix: log debug message when it is not in single node mode;